app/Http/Controllers/Table/PortsController.php

```diff
@@ -125,7 +125,7 @@ protected function baseQuery($request)
             ->when($request->get('state'), function (Builder $query, $state) {
                 switch ($state) {
                     case 'down':
-                        return $query->where('ifAdminStatus', 'up')->where('ifOperStatus', 'down');
+                        return $query->where('ifAdminStatus', 'up')->where('ifOperStatus', '!=', 'up');
```

Comment:
There are scopes too use for this instead to keep things tidy.  https://github.com/librenms/librenms/blob/master/app/Models/Port.php#L202

Reply:
How do I use that here as ` return $query->isDown();` doesn't return any ports

Reply:
There is a join mucking up the bit:
```
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'ignore' in where clause is ambiguous (Connection: mysql, SQL: select count(*) as aggregate from `ports` left join `devices` on `ports`.`device_id` = `devices`.`device_id` where `deleted` = 0 and (`deleted` = 0 and `ignore` = 0 and `disabled` = 0 and `ifOperStatus` != up and `ifAdminStatus` = up) and `ports`.`disabled` = 0 and `ports`.`ignore` = 0)
```

Reply:
The fix for the scopes:
```php
    // ---- Accessors/Mutators ----

    public function getIfPhysAddressAttribute($mac)
    {
        if (! empty($mac)) {
            return preg_replace('/(..)(..)(..)(..)(..)(..)/', '\\1:\\2:\\3:\\4:\\5:\\6', $mac);
        }

        return null;
    }

    // ---- Query scopes ----

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsDeleted($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), 1],
        ]);
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsNotDeleted($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), 0],
        ]);
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsUp($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), '=', 0],
            [$this->qualifyColumn('ignore'), '=', 0],
            [$this->qualifyColumn('disabled'), '=', 0],
            [$this->qualifyColumn('ifOperStatus'), '=', 'up'],
        ]);
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsDown($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), '=', 0],
            [$this->qualifyColumn('ignore'), '=', 0],
            [$this->qualifyColumn('disabled'), '=', 0],
            [$this->qualifyColumn('ifOperStatus'), '!=', 'up'],
            [$this->qualifyColumn('ifAdminStatus'), '=', 'up'],
        ]);
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsShutdown($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), '=', 0],
            [$this->qualifyColumn('ignore'), '=', 0],
            [$this->qualifyColumn('disabled'), '=', 0],
            [$this->qualifyColumn('ifAdminStatus'), '=', 'down'],
        ]);
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsIgnored($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), '=', 0],
            [$this->qualifyColumn('ignore'), '=', 1],
        ]);
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsDisabled($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), '=', 0],
            [$this->qualifyColumn('disabled'), '=', 1],
        ]);
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeHasErrors($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), '=', 0],
            [$this->qualifyColumn('ignore'), '=', 0],
            [$this->qualifyColumn('disabled'), '=', 0],
        ])->where(function ($query) {
            /** @var Builder $query */
            $query->where($this->qualifyColumn('ifInErrors_delta'), '>', 0)
                ->orWhere($this->qualifyColumn('ifOutErrors_delta'), '>', 0);
        });
    }

    /**
     * @param  Builder  $query
     * @return Builder
     */
    public function scopeIsValid($query)
    {
        return $query->where([
            [$this->qualifyColumn('deleted'), '=', 0],
            [$this->qualifyColumn('disabled'), '=', 0],
        ]);
    }
```

Reply:
Which makes this work:
```php
            ->when($request->get('errors'), function (Builder $query) {
                return $query->hasErrors();
            })
            ->when($request->get('state'), function (Builder $query, $state) {
                return match ($state) {
                    'down' => $query->isDown(),
                    'up' => $query->isUp(),
                    'admindown' => $query->isShutdown(),
                    default => $query,
                };
            });
```

Reply:
The menu counts already use the scopes :)